x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/knowledge_base/get_knowledge_base_entry.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient, Logger } from '@kbn/core/server';
+import { KnowledgeBaseEntryResponse } from '@kbn/elastic-assistant-common';
+import { AuthenticatedUser } from '@kbn/security-plugin/common';
+import { EsKnowledgeBaseEntrySchema } from './types';
+import { transformESSearchToKnowledgeBaseEntry } from './transforms';
+
+export interface GetKnowledgeBaseEntryParams {
+  esClient: ElasticsearchClient;
+  logger: Logger;
+  knowledgeBaseIndex: string;
+  id: string;
+  user?: AuthenticatedUser | null;
+}
+
+export const getKnowledgeBaseEntry = async ({
+  esClient,
+  logger,
+  knowledgeBaseIndex,
+  id,
+  user,
+}: GetKnowledgeBaseEntryParams): Promise<KnowledgeBaseEntryResponse | null> => {
+  const filterByUser = user
+    ? [
+        {
+          nested: {
+            path: 'users',
+            query: {
+              bool: {
+                must: [
+                  {
+                    match: user.profile_uid
+                      ? { 'users.id': user.profile_uid }
+                      : { 'users.name': user.username },
+                  },
+                ],
+              },
+            },
+          },
+        },
+      ]
+    : [];
+  try {
+    const response = await esClient.search<EsKnowledgeBaseEntrySchema>({
+      query: {
+        bool: {
+          must: [
+            {
+              bool: {
+                should: [
+                  {
+                    term: {
+                      _id: id,
+                    },
+                  },
+                ],
+              },
+            },
+            ...filterByUser,
```

Comment:
Is `...filterByUser` effectively a `noop` when

```typescript
user?: AuthenticatedUser | null
```

 is `null`? Consider throwing an error if `AuthenticatedUser` is `null`

Reply:
Good suggestion, thank you! See the response in the comment above: https://github.com/elastic/kibana/pull/182763#discussion_r1605432444

Reply:
Ended up making `AuthenticatedUser` required here

Reply:
Similar to the above question - how we will handle non-user owned (shared) knowledge resources?